cylc/uiserver/tests/test_workflow_retrieval.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import pytest
+import sqlite3
+
+from cylc.flow.id import Tokens
+from cylc.uiserver.schema import make_query
+
+'''This file tests the ability for the cylc UI to retrieve workflow information
+and perform simple statistical calculations for the analysis tab'''
+
+
+def test_make_query_1():
+    conn = sqlite3.connect(':memory:')
+    conn.execute('''CREATE TABLE task_jobs(cycle TEXT, name TEXT,
+    submit_num INTEGER, flow_nums TEXT, is_manual_submit INTEGER,
+    try_num INTEGER, time_submit TEXT, time_submit_exit TEXT,
+    submit_status INTEGER, time_run TEXT, time_run_exit TEXT,
+    run_signal TEXT, run_status INTEGER, platform_name TEXT,
+    job_runner_name TEXT, job_id TEXT,
+    PRIMARY KEY(cycle, name, submit_num));''')
+
+    conn.executemany(
+        'INSERT into task_jobs VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)',
+        [('1', 'Task_1', '01', '{1}', 0, 1,
+          '2022-12-14T15:00:00Z', '2022-12-14T15:01:00Z', 0,
+          '2022-12-14T15:01:00Z', '2022-12-14T15:10:00Z', None, 0,
+          'MyPlatform', 'User', 'UsersJob')
+         ])
+    conn.commit()
+    workflow = Tokens('~user/workflow')
+
+    return_value = make_query(conn, workflow)
```

Comment:
Can avoid creating the table manually by using our good friend the `CylcWorkflowDAO`
```suggestion
    with CylcWorkflowDAO(':memory:', create_tables=True) as dao:
        conn = dao.connect()
        conn.executemany(
            'INSERT into task_jobs VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)',
            [('1', 'Task_1', '01', '{1}', 0, 1,
              '2022-12-14T15:00:00Z', '2022-12-14T15:01:00Z', 0,
              '2022-12-14T15:01:00Z', '2022-12-14T15:10:00Z', None, 0,
              'MyPlatform', 'User', 'UsersJob')
             ])
        conn.commit()
        workflow = Tokens('~user/workflow')

        return_value = make_query(conn, workflow)
```

Reply:
@MetRonnie, this was purposeful.

The UIS is going to have to maintain a range of cylc-flow version compatibility. This approach ensures there is a test breakage when DB structure changes and allows us to mock DBs from multiple cylc-flow versions.

Reply:
I would think manually creating the table will **not** see a breakage when the DB structure changes, whereas using the automatic table creation will? Can discuss at today's catch-up

Reply:
> I would think manually creating the table will not see a breakage when the DB structure changes

Correct, but to ensure compatibility we would want both.

Reply:
Punting to a different issue: #441